crates/ra_analysis/src/imp.rs

```diff
@@ -184,6 +184,28 @@ impl AnalysisImpl {
         };
         Ok(query.search(&buf))
     }
+
+    pub(crate) fn module_path(&self, position: FilePosition) -> Cancelable<String> {
```

Comment:
I feel like this should return `Cancelable<Option<String>>` as opposed to returning `Ok("")` in the failure cases.